### ISSUE TITLE
python harp.concatenate fails when products have integer variable with different size in vertical

### ISSUE THREAD

**Issue by StevenCompernolle** [stcorp/harp#308]
`harp.concatenate` fails when products have integer variable with different size in the vertical dimension.

```
.../py39/lib64/python3.9/site-packages/harp/_harppy.py in concatenate(products)
   1501                 for i in range(len(target_variable.data.shape))[1:]:
   1502                     if target_variable.data.shape[i] < source_variable.data.shape[i]:
-> 1503                         _extend_variable_for_dim(target_variable, i, source_variable.data.shape[i])
   1504                     if source_variable.data.shape[i] < target_variable.data.shape[i]:
   1505                         _extend_variable_for_dim(source_variable, i, target_variable.data.shape[i])

...//py39/lib64/python3.9/site-packages/harp/_harppy.py in _extend_variable_for_dim(variable, dim_index, new_length)
   1439     shape[dim_index] = new_length - shape[dim_index]
   1440     filler = numpy.empty(shape, dtype=variable.data.dtype)
-> 1441     filler[:] = numpy.NAN
   1442     variable.data = numpy.concatenate([variable.data, filler], axis=dim_index)
   1443 

ValueError: cannot convert float NaN to integer
```
The cause seems that aligning vertical dimension should be done by padding with NaN, but this is not possible for an integer variable.

Note that `harp.execute_operations` (and probably also commandline `harpmerge`) succeeds in this case. It seems that padding of an integer variable is done with zero here. 
This could be seen as a problem, since zero can be a perfectly valid value. 
But normally one expects an `altitude` or `pressure` variable to be present in the product as well, which should then be aligned by padding with NaN, pointing to the user the invalidity of these values.

So the simplest solution is that `harp.concatenate` extends by padding with zero (to mimick `harp.execute_operations` and `harpmerge` behaviour).

A more involved solution that prevents padding with valid values would be that a `_Fillvalue` is defined for integer variables with vertical dimension, but then the user should have a mean to set a `_Fillvalue` before doing the `harp.concatenate` or `harpmerge`.

**Comment by svniemeijer** [comment]
This is indeed a bug that we will fix. The python implementation of the concatenate functionality should behave the same as the C implementation.

At the moment the 'fill values' in HARP are indeed defined as [NaN for floating point values and 0 for integer values](https://github.com/stcorp/harp/blob/54f5aaaa68fdd8d1c1e8a343a424f668c37566b4/libharp/harp-utils.c#L537). The reason behind this is that for floating point values NaN is the natural 'invalid value' indicator, but for integers there exists no such value. Using approaches such as MIN_INT or MAX_INT will create problems when trying to convert from one integer type to the other; ideally the fill value stays the same, independent of the byte size of the integer. This is why in HARP we've opted for using a 0 value here. And indeed, this is then not an invalid value (we will also have to update [the documentation](http://stcorp.github.io/harp/doc/html/conventions/dimensions.html) to reflect this). A user will have to look at an associated floating point variable (such as the axis variable) to determine the validity.

Note that using `_Fillvalue` is also still not guaranteed to mean an invalid value, so an approach based on that attribute would not be a good choice. The validity of a value in a variable is purely [determined by the `valid_min` and `valid_max` attributes](http://stcorp.github.io/harp/doc/html/conventions/variable_attributes.html), but that means that in most cases (when valid min/max does not align with the min/max of the data type) there are many values that can be used to indicate 'invalid', but only for floating point values we have a natural choice, which is NaN.

**Comment by svniemeijer** [comment]
Fixed in 8c7864e66025b9c87914e764856a960fff127c26
